### PR TITLE
feat(bar chart): sort bars add asc and desc

### PR DESCRIPTION
**Pull request by zhaoyu89730105**
Sort bars add asc and desc.
wx20180626-130941 2x

### CODE REVIEW

File: superset/assets/src/explore/controls.jsx [hunk 1]
```diff
@@ -449,11 +449,16 @@ export const controls = {
...
-    type: 'CheckboxControl',
+    type: 'SelectControl',
     label: t('Sort Bars'),
-    default: false,
-    renderTrigger: true,
+    default: 'unsort',
     description: t('Sort bars by x labels.'),
+    renderTrigger: true,
+    choices: formatSelectOptions([
+        'unsort',
```
Comment (hughhhh) [inline superset/assets/src/explore/controls.jsx]: Why are allowing an unsort option? We should just pick `asc` or `desc` as a default and then allow the user to toggle

Comment (zhaoyu89730105) [reply superset/assets/src/explore/controls.jsx]: The cancellation sort will be displayed in the original order of the data.

File: superset/assets/src/visualizations/nvd3_vis.js [hunk 2]
```diff
@@ -249,11 +249,17 @@ export default function nvd3Vis(slice, payload) {
...
-          data.forEach((d) => {
-            d.values.sort((a, b) => tryNumify(a.x) < tryNumify(b.x) ? -1 : 1);
-          });
+        if (fd.order_bars === "asc") {
+            data.forEach((d) => {
+                d.values.sort((a, b) => tryNumify(a.x) < tryNumify(b.x) ? -1 : 1);
+            });
         }
+        if (fd.order_bars === "desc") {
+            data.forEach((d) => {
```
Comment (mistercrunch) [inline superset/assets/src/visualizations/nvd3_vis.js]: `d.values.sort.reverse()` maybe instead of repeating the code

**Review by mistercrunch — CHANGES_REQUESTED** [review]
I think you also need a db migration script to reassign all the existing charts to the proper option

**Pull request closed without merging**